service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java

```diff
@@ -107,11 +135,10 @@ private void logCacheGetDebugMessages(final PolicyEvaluationRequestCacheKey key,
     @Override
     public void set(final PolicyEvaluationRequestCacheKey key, final PolicyEvaluationResult result) {
         try {
-            DateTime now = new DateTime();
-            result.setTimestamp(now.getMillis());
+            setEntityTimestamps(key, result);
+            result.setTimestamp(new DateTime().getMillis());
             String value = OBJECT_MAPPER.writeValueAsString(result);
```

Comment:
Add a unit test that caches the eval result with resolved URIs and get returns the valid cache since timestamps are recorded.